docling/datamodel/document.py

```diff
@@ -264,6 +264,11 @@ def _guess_format(self, obj: Union[Path, DocumentStream]):
                 with obj.open("rb") as f:
                     content = f.read(1024)  # Read first 1KB
 
+            # Detect PubMed XML documents
+            xml_doctype = re.search(r"<!DOCTYPE [^>]+>", content.decode("utf-8"))
+            if xml_doctype and ("/NLM//DTD JATS" in xml_doctype.group()):
+                return InputFormat.XML_PUBMED
+
```

Comment:
The PubMed content won't get detected if passed as a `DocumentStream`. We should move this further down.